## Author : Mercy Njambi Mukundi

# *1. Introduction*

##  1.1 Overview

## 1.2 Objectives

## 1.3 Questions to answer

1. What type of films are currently doing the best at the box office?


# *2. Data wrangling*

## 2.1 Data Description

## 2.2 Loading Libraries and Datasets

In [1]:
# importing the libraries

import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [57]:
# reading the datasets into a dataframe

df = pd.read_csv("DATA/bom.movie_gross.csv")
df_budget = pd.read_csv("DATA/tn.movie_budgets.csv", index_col=0)
df_ratings = pd.read_csv('DATA/tmdb.movies.csv', index_col=0)
df_genres = pd.read_csv("DATA/title.basics.csv")

### 2.2.1 Previewing the Datasets

In [3]:
# previewing the first dataset

df.head(3)

title studio  domestic_gross  \
0                                  Toy Story 3     BV     415000000.0   
1                   Alice in Wonderland (2010)     BV     334200000.0   
2  Harry Potter and the Deathly Hallows Part 1     WB     296000000.0   

  foreign_gross  year  
0     652000000  2010  
1     691300000  2010  
2     664300000  2010

In [4]:
# previewing the second dataset

df_budget.head(3)

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   

   production_budget domestic_gross worldwide_gross  
id                                                   
1       $425,000,000   $760,507,625  $2,776,345,279  
2       $410,600,000   $241,063,875  $1,045,663,875  
3       $350,000,000    $42,762,350    $149,762,350

In [5]:
# previewing the third datasets

df_ratings.head(3)

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368

In [62]:
# previewing the fourth datasets

df_genres.head(3)

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   

   start_year  runtime_minutes              genres  
0        2013            175.0  Action,Crime,Drama  
1        2019            114.0     Biography,Drama  
2        2018            122.0               Drama

### Accessing Information About the Datasets

In [6]:
# Accessing more information about the first dataset

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3387 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3387 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3359 non-null   float64
 3   foreign_gross   2037 non-null   object 
 4   year            3387 non-null   int64  
dtypes: float64(1), int64(1), object(3)
memory usage: 132.4+ KB


This shows that data has 3 data types: floats, intergers and objects. The columns title, studio and foreign gross are of type object, the year column is of type integer, and domestic gross is of the float type.

In [7]:
df.shape

(3387, 5)

This shows the data has 3,387 rows and 5 columns

In [8]:
# Accessing more information about the second dataset

df_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   release_date       5782 non-null   object
 1   movie              5782 non-null   object
 2   production_budget  5782 non-null   object
 3   domestic_gross     5782 non-null   object
 4   worldwide_gross    5782 non-null   object
dtypes: object(5)
memory usage: 271.0+ KB


All data has the data type object

In [9]:
df_budget.shape

(5782, 5)

This shows the data has 5,782 rows and 5 columns

In [35]:
# Accessing more information about the third dataset

df_ratings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


This shows that the data has 3 data types: floats, intergers and objects. The columns genre ids, original language, original title, release date and title are of type object, the id and vote count columns are of type integer, and popularity and vote average are of the float type.

In [36]:
df_ratings.shape

(26517, 9)

This shows the data has 26,517 rows and 9 columns

In [63]:
#Accessing more information about the fourth dataset

df_genres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


This shows that data has 3 data types: floats, intergers and objects. The columns tconst, primary title, original title and genres are of type object, the start year column is of type integer, and runtime minutes is of the float type.

In [60]:
df_genres.shape

(146144, 6)

This shows the data has 146,144 rows and 6 columns

## 2.3 Data Cleaning

### 2.3.1 Movie Studios Dataset

From the preview of the data, it is noted that the column 'foreign_gross' stores numbers, but it shown to be of data type object in the dataframe. This needs to be changed to an number, to enable analysis.

In [10]:
df.loc[df['foreign_gross'].fillna('0').str.contains(','),['foreign_gross']]

foreign_gross
1872       1,131.6
1873       1,019.4
1874       1,163.0
2760       1,010.0
3079       1,369.5

Upon investigation of the column foreign gross, it is noted that there are some rows with commas in the data. Since this column is to be converted to an integer, this needs to be remedied. Since the column has some null values, the column will be converted to a float type.

In [11]:
# converting the column to a float

df['foreign_gross'] = df['foreign_gross'].map(lambda x: str(x).replace(',','')).astype('float64')

In [12]:
#c checking they data type of foreign gross column

df['foreign_gross'].dtypes

dtype('float64')

Now that the data columns have the appropriate data types, the data can now be checked for missing and duplicate values.

In [13]:
# checking for columns with   missing data

df.isna().sum()

title                0
studio               5
domestic_gross      28
foreign_gross     1350
year                 0
dtype: int64

This data has null values in 3 columns: 'studio' with 5, 'domestic_gross' with 28 and 'foreign_gross' with 1350.

In [14]:
# checking for duplicates in the data

df.duplicated().sum()

0

The data has no duplicates.

In [15]:
# accessing the rows of studio column with null values

df[df['studio'].isnull()]

title studio  domestic_gross  foreign_gross  \
210   Outside the Law (Hors-la-loi)    NaN         96900.0      3300000.0   
555         Fireflies in the Garden    NaN         70600.0      3300000.0   
933           Keith Lemon: The Film    NaN             NaN      4000000.0   
1862                 Plot for Peace    NaN          7100.0            NaN   
2825               Secret Superstar    NaN             NaN    122000000.0   

      year  
210   2010  
555   2011  
933   2012  
1862  2014  
2825  2017

Since a section of the analysis depends on the studios column data, it makes no sense to continue with the rows under studio that have no data, hence, they need to be droppped. There are 5 rows with null values. This number is small enough, and won't distort our outcome.

In [16]:
# dropping the rows under studio column with null values

df.dropna(how = 'any', subset = ['studio'],inplace=True)

In [17]:
# checking to see if the above changes took effect

df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3356 non-null   float64
 3   foreign_gross   2033 non-null   float64
 4   year            3382 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 158.5+ KB


It can be noted that the number of rows has reduced by 5, to 3382. The rest on the null values can be replaced with a 0 in the other columns(domestic and foreign gross). In the domestic gross, the number of null values are small, hence won't distort our outcome. On the other hand, the foreign column has alot of null values;1349 null, out of 3382 rows. This is about 40% of the rows. This is a very high number, and should be investigated later on in this analysis. For now, we can replace the null values with 0.

In [18]:
# replacing the missing values with 0

df.fillna(0, inplace = True)

In [19]:
#checking our data for null entries.
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3382 entries, 0 to 3386
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   title           3382 non-null   object 
 1   studio          3382 non-null   object 
 2   domestic_gross  3382 non-null   float64
 3   foreign_gross   3382 non-null   float64
 4   year            3382 non-null   int64  
dtypes: float64(2), int64(1), object(2)
memory usage: 158.5+ KB


It can be noted that there are no null values in our data now. 

### 2.3.2 Movie Budget Dataset

In [20]:
df_budget.head(3)

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   

   production_budget domestic_gross worldwide_gross  
id                                                   
1       $425,000,000   $760,507,625  $2,776,345,279  
2       $410,600,000   $241,063,875  $1,045,663,875  
3       $350,000,000    $42,762,350    $149,762,350

In [21]:
df_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   release_date       5782 non-null   object
 1   movie              5782 non-null   object
 2   production_budget  5782 non-null   object
 3   domestic_gross     5782 non-null   object
 4   worldwide_gross    5782 non-null   object
dtypes: object(5)
memory usage: 271.0+ KB


It can be noted that the columns 'production_budget', 'domestic_gross' and 'worldwide_gross' have numerical data, but are of object type object. This can be attributed to them having the '$' symbol and ','. These need to be removed, so that the columns to be converted to integers.

In [22]:
# def remove_symbols(data,column,'symbol1','symbol2','replace1','replace2'):
#     result = data['column'].replace('symbol1','replace1').replace('symbol2','replace2')
    
#     return result  

In [23]:
# remove_symbols(df_budget,production_budget,'$',',','','')

In [24]:
# removing the $ and , from the columns

df_budget['production_budget'] = df_budget['production_budget'].str.replace('$','')
df_budget['production_budget'] = df_budget['production_budget'].str.replace(',', '')
df_budget['domestic_gross'] = df_budget['domestic_gross'].str.replace('$','')
df_budget['domestic_gross'] = df_budget['domestic_gross'].str.replace(',', '')
df_budget['worldwide_gross'] = df_budget['worldwide_gross'].str.replace('$','')
df_budget['worldwide_gross'] = df_budget['worldwide_gross'].str.replace(',', '')

In [25]:
# converting the columns to integers

df_budget['production_budget'] = df_budget['production_budget'].astype('int64')
df_budget['domestic_gross'] = df_budget['domestic_gross'].astype('int64')
df_budget['worldwide_gross'] = df_budget['worldwide_gross'].astype('int64')

In [26]:
df_budget

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
..           ...                                          ...   
78  Dec 31, 2018                                       Red 11   
79   Apr 2, 1999                                    Following   
80  Jul 13, 2005                Return to the Land of Wonders   
81  Sep 29, 2015                         A Plague So Pleasant   
82   Aug 5, 2005                            My Date With Drew   

    production_budget  domestic_gross  worldwide_gross  
id                                                      
1           425000000       760507625       2776345279  
2           410600000       241063875       1045663875  
3           350000000        42762350        149762350  
4           330600000       459005868       1403013963  
5           317000000       620181382       1316721747  
..                ...             ...              ...  
78               7000               0                0  
79               6000           48482           240495  
80               5000            1338             1338  
81               1400               0                0  
82               1100          181041           181041  

[5782 rows x 5 columns]

In [27]:
#checking the data types of the columns

df_budget.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5782 entries, 1 to 82
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   release_date       5782 non-null   object
 1   movie              5782 non-null   object
 2   production_budget  5782 non-null   int64 
 3   domestic_gross     5782 non-null   int64 
 4   worldwide_gross    5782 non-null   int64 
dtypes: int64(3), object(2)
memory usage: 271.0+ KB


Now that the data columns have the appropriate data types, the data can now be checked for missing and duplicate values.

In [28]:
# checking if the data has missing values

df_budget.isna().sum()

release_date         0
movie                0
production_budget    0
domestic_gross       0
worldwide_gross      0
dtype: int64

The dataset has no null values

In [29]:
# checking for duplicate values

df_budget.duplicated().sum()

0

The data has no duplicates

### 2.3.3 Movie Ratings

In [37]:
df_ratings.head(3)

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   
2        [12, 28, 878]  10138                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   
2                                    Iron Man 2      28.515   2010-05-07   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610  
2                                    Iron Man 2           6.8       12368

In [40]:
# checking the data types of the columns

df_ratings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 26517 entries, 0 to 26516
Data columns (total 9 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   genre_ids          26517 non-null  object 
 1   id                 26517 non-null  int64  
 2   original_language  26517 non-null  object 
 3   original_title     26517 non-null  object 
 4   popularity         26517 non-null  float64
 5   release_date       26517 non-null  object 
 6   title              26517 non-null  object 
 7   vote_average       26517 non-null  float64
 8   vote_count         26517 non-null  int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 2.0+ MB


From analysis, the data types of the different columns coincide with the data they hold.

Now that the data columns have the appropriate data types, the data can now be checked for missing and duplicate values.

In [41]:
# checking if the data has missing values

df_ratings.isna().sum()

genre_ids            0
id                   0
original_language    0
original_title       0
popularity           0
release_date         0
title                0
vote_average         0
vote_count           0
dtype: int64

The data has no missing values

In [49]:
# checking for duplicate values

df_ratings.duplicated().sum()
#df_ratings[df_ratings.duplicated(['genre_ids','id','original_title','popularity'])]
#df_ratings[df_ratings['id'] == 862]

1020

This dataset has 1020 duplicates. These rows need to be dropped.

In [53]:
# dropping the duplicates row

df_ratings.drop_duplicates(inplace=True)

In [55]:
df_ratings.shape

(25497, 9)

In [56]:
df_ratings

genre_ids      id original_language  \
0          [12, 14, 10751]   12444                en   
1      [14, 12, 16, 10751]   10191                en   
2            [12, 28, 878]   10138                en   
3          [16, 35, 10751]     862                en   
4            [28, 878, 12]   27205                en   
...                    ...     ...               ...   
26512             [27, 18]  488143                en   
26513             [18, 53]  485975                en   
26514         [14, 28, 12]  381231                en   
26515      [10751, 12, 28]  366854                en   
26516             [53, 27]  309885                en   

                                     original_title  popularity release_date  \
0      Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                          How to Train Your Dragon      28.734   2010-03-26   
2                                        Iron Man 2      28.515   2010-05-07   
3                                         Toy Story      28.005   1995-11-22   
4                                         Inception      27.920   2010-07-16   
...                                             ...         ...          ...   
26512                         Laboratory Conditions       0.600   2018-10-13   
26513                               _EXHIBIT_84xxx_       0.600   2018-05-01   
26514                                  The Last One       0.600   2018-10-01   
26515                                  Trailer Made       0.600   2018-06-22   
26516                                    The Church       0.600   2018-10-05   

                                              title  vote_average  vote_count  
0      Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                          How to Train Your Dragon           7.7        7610  
2                                        Iron Man 2           6.8       12368  
3                                         Toy Story           7.9       10174  
4                                         Inception           8.3       22186  
...                                             ...           ...         ...  
26512                         Laboratory Conditions           0.0           1  
26513                               _EXHIBIT_84xxx_           0.0           1  
26514                                  The Last One           0.0           1  
26515                                  Trailer Made           0.0           1  
26516                                    The Church           0.0           1  

[25497 rows x 9 columns]

### 2.3.4 Movie Genres Dataset

In [64]:
df_genres.head()

tconst                    primary_title              original_title  \
0  tt0063540                        Sunghursh                   Sunghursh   
1  tt0066787  One Day Before the Rainy Season             Ashad Ka Ek Din   
2  tt0069049       The Other Side of the Wind  The Other Side of the Wind   
3  tt0069204                  Sabse Bada Sukh             Sabse Bada Sukh   
4  tt0100275         The Wandering Soap Opera       La Telenovela Errante   

   start_year  runtime_minutes                genres  
0        2013            175.0    Action,Crime,Drama  
1        2019            114.0       Biography,Drama  
2        2018            122.0                 Drama  
3        2018              NaN          Comedy,Drama  
4        2017             80.0  Comedy,Drama,Fantasy

In [65]:
# checking the data type of the datasets

df_genres.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 146144 entries, 0 to 146143
Data columns (total 6 columns):
 #   Column           Non-Null Count   Dtype  
---  ------           --------------   -----  
 0   tconst           146144 non-null  object 
 1   primary_title    146144 non-null  object 
 2   original_title   146123 non-null  object 
 3   start_year       146144 non-null  int64  
 4   runtime_minutes  114405 non-null  float64
 5   genres           140736 non-null  object 
dtypes: float64(1), int64(1), object(4)
memory usage: 6.7+ MB


All the column's data types correspond to the data in the columns.

Now that the data columns have the appropriate data types, the data can now be checked for missing and duplicate values.

In [66]:
# checking for missing values

df_genres.isna().sum()

tconst                 0
primary_title          0
original_title        21
start_year             0
runtime_minutes    31739
genres              5408
dtype: int64

The columns original title, runtime minutes and genres all have missing values. Original title has 21 missing values, runtime minutes 31,739 and genres has 5,408 missing values.

In [67]:
# checking for duplicates

df_genres.duplicated().sum()

0

The data has no duplicate rows

# *3.0 Exploratory Data Analysis (EDA)*

## 3.1 Movie Studios Dataset

In [31]:
# creating a new column for worldwide gross

df['total_gross'] = df['domestic_gross'] + df['foreign_gross']
df

title      studio  domestic_gross  \
0                                     Toy Story 3          BV     415000000.0   
1                      Alice in Wonderland (2010)          BV     334200000.0   
2     Harry Potter and the Deathly Hallows Part 1          WB     296000000.0   
3                                       Inception          WB     292600000.0   
4                             Shrek Forever After        P/DW     238700000.0   
...                                           ...         ...             ...   
3382                                    The Quake       Magn.          6200.0   
3383                  Edward II (2018 re-release)          FM          4800.0   
3384                                     El Pacto        Sony          2500.0   
3385                                     The Swan  Synergetic          2400.0   
3386                            An Actor Prepares       Grav.          1700.0   

      foreign_gross  year   total_gross  
0       652000000.0  2010  1.067000e+09  
1       691300000.0  2010  1.025500e+09  
2       664300000.0  2010  9.603000e+08  
3       535700000.0  2010  8.283000e+08  
4       513900000.0  2010  7.526000e+08  
...             ...   ...           ...  
3382            0.0  2018  6.200000e+03  
3383            0.0  2018  4.800000e+03  
3384            0.0  2018  2.500000e+03  
3385            0.0  2018  2.400000e+03  
3386            0.0  2018  1.700000e+03  

[3382 rows x 6 columns]

In [32]:
df.groupby(['studio', 'year']).mean()

domestic_gross  foreign_gross  total_gross
studio year                                            
3D     2010       6100000.0      9900000.0   16000000.0
A23    2011        151000.0            0.0     151000.0
       2013         13200.0            0.0      13200.0
A24    2013       5569080.0      6180000.0   11749080.0
       2014       1898830.0      1360000.0    3258830.0
...                     ...            ...          ...
Zeit.  2013        392500.0            0.0     392500.0
       2014        126175.0       925000.0    1051175.0
       2015         22900.0            0.0      22900.0
       2016         11700.0            0.0      11700.0
       2017        820000.0            0.0     820000.0

[742 rows x 3 columns]

## 3.2 Movie Budget Dataset

Using the assumption that profits made on a movie is the difference between the total gross and the production budget, the profits for the dataset are calculated.

In [33]:
df_budget.head(3)

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   

    production_budget  domestic_gross  worldwide_gross  
id                                                      
1           425000000       760507625       2776345279  
2           410600000       241063875       1045663875  
3           350000000        42762350        149762350

In [34]:
# calculating the profit per movie

df_budget['movie_profits'] = df_budget['worldwide_gross'] - df_budget['production_budget']
df_budget

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   
3    Jun 7, 2019                                 Dark Phoenix   
4    May 1, 2015                      Avengers: Age of Ultron   
5   Dec 15, 2017            Star Wars Ep. VIII: The Last Jedi   
..           ...                                          ...   
78  Dec 31, 2018                                       Red 11   
79   Apr 2, 1999                                    Following   
80  Jul 13, 2005                Return to the Land of Wonders   
81  Sep 29, 2015                         A Plague So Pleasant   
82   Aug 5, 2005                            My Date With Drew   

    production_budget  domestic_gross  worldwide_gross  movie_profits  
id                                                                     
1           425000000       760507625       2776345279     2351345279  
2           410600000       241063875       1045663875      635063875  
3           350000000        42762350        149762350     -200237650  
4           330600000       459005868       1403013963     1072413963  
5           317000000       620181382       1316721747      999721747  
..                ...             ...              ...            ...  
78               7000               0                0          -7000  
79               6000           48482           240495         234495  
80               5000            1338             1338          -3662  
81               1400               0                0          -1400  
82               1100          181041           181041         179941  

[5782 rows x 6 columns]

In [68]:
df.head(2)

title studio  domestic_gross foreign_gross  year
0                 Toy Story 3     BV     415000000.0     652000000  2010
1  Alice in Wonderland (2010)     BV     334200000.0     691300000  2010

In [69]:
df_budget.head(2)

release_date                                        movie  \
id                                                              
1   Dec 18, 2009                                       Avatar   
2   May 20, 2011  Pirates of the Caribbean: On Stranger Tides   

   production_budget domestic_gross worldwide_gross  
id                                                   
1       $425,000,000   $760,507,625  $2,776,345,279  
2       $410,600,000   $241,063,875  $1,045,663,875

In [70]:
df_ratings.head(2)

genre_ids     id original_language  \
0      [12, 14, 10751]  12444                en   
1  [14, 12, 16, 10751]  10191                en   

                                 original_title  popularity release_date  \
0  Harry Potter and the Deathly Hallows: Part 1      33.533   2010-11-19   
1                      How to Train Your Dragon      28.734   2010-03-26   

                                          title  vote_average  vote_count  
0  Harry Potter and the Deathly Hallows: Part 1           7.7       10788  
1                      How to Train Your Dragon           7.7        7610

In [71]:
df_genres.head(2)

tconst                    primary_title   original_title  start_year  \
0  tt0063540                        Sunghursh        Sunghursh        2013   
1  tt0066787  One Day Before the Rainy Season  Ashad Ka Ek Din        2019   

   runtime_minutes              genres  
0            175.0  Action,Crime,Drama  
1            114.0     Biography,Drama

In [75]:
b_r = (df_budget['movie']).isin(df_ratings['title'])

In [76]:
b_r

id
1      True
2      True
3     False
4      True
5     False
      ...  
78    False
79    False
80    False
81    False
82    False
Name: movie, Length: 5782, dtype: bool